flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,378 @@
+            if quality['allowed']:
+                name = quality['quality']['name']
+                quality_req = radarr_quality_to_flexget_quality_req(name)
+                if quality_req:
+                    quality_requirements.append(quality_req)
+
+    return quality_requirements
+
+
+
+class RadarrSet(MutableSet):
+    """ Accesses the Radarr movies using the provided the config """
+
+    def __init__(self, config):
+        self.config = config
+        self.service = RadarrAPIService(
+            config['api_key'],
+            config['base_url'],
+            config['port']
+            )
+
+        # Class member used for caching the items to avoid
+        # unnecessary calls to the Radarr API.
+        # We use the self.items property to access it.
+        # Just set this to None again later to invalidate the cache.
+        self._movie_entries = None
+
+    def __iter__(self):
+        return iter(self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def discard(self, entry):
+        if not entry:
+            return
+
+        matching_entry = self.__find_matching_entry(entry)
+        if matching_entry:
+            movie_id = matching_entry['radarr_id']
+            response = self.service.delete_movie(movie_id)
+            log.verbose('Removed movie %s from Radarr', matching_entry['title'])
+            # Clear the cache
+            self._movie_entries = None
+        else:
+            log.debug('Could not find any matching movie to remove for entry %s', entry)
+
+    def __ior__(self, other):
+        for entry in other:
+            self.add(entry)
+
+    def __contains__(self, entry):
+        if not entry:
+            return False
+
+        matching_entry = self.__find_matching_entry(entry)
+
+        return matching_entry is not None
+
+    def add(self, entry):
+
+        # The easiest way to add a movie to Radarr is to
+        # first use the lookup API. Using that we will get
+        # a json response which gives us most of the input
+        # we need for the POST request.
+
+        result = self.__lookup_movie(entry.get('title'),
+                                     entry.get('imdb_id'),
+                                     entry.get('tmdb_id'))
+
+        if result:
+            rootFolders = self.service.get_root_folders()
+            rootFolderPath = rootFolders[0]['path']
+
+            # TODO: should we let the user affect this one,
+            # or try to parse the 'quality' entry somehow?
+            qualityProfileId = 1
+
+            try:
+                add_result = self.service.add_movie(
+                    result['title'],
+                    qualityProfileId,
+                    result['titleSlug'],
+                    result['images'],
+                    result['tmdbId'],
+                    rootFolderPath)
+                log.verbose('Added movie %ls to Radarr list', result['title'])
+            except RadarrMovieAlreadyExistsError:
```

Comment:
The sonarr_list plugin checks if the entry is already added by first fetching the list from sonarr and does not fail if that is this case.  Do we really want to behave different here?